exchange/bidder.go

```diff
@@ -243,6 +261,15 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb2.B
 					}
 				}
 
+				if len(bidderRequest.BidderStoredResponses) > 0 {
+					//set imp ids back to response for bids with stored responses
+					for i := 0; i < len(bidResponse.Bids); i++ {
+						if httpInfo.request.Uri == "" {
+							bidResponse.Bids[i].Bid.ImpID = string(httpInfo.request.Body)
+						}
+					}
+				}
```

Comment:
What is happening here?

Reply:
Discussed offline. 
Short idea: For stored bid response in order to imitate real response and keep track of imp id, I put imp id to request body. I was thinking about adding it to response itself, but different bidders may have different response formats, so it can be a name collision and it will require extra byte array manipulations (put this value and then extract it back for all responses, even for real ones) so I decided to put it to request body - it's empty for stored responses anyways. For stored responses URL is also empty and this is a reliable identifier of stored response. It's easy to implement and response remains without any modifications.